frontend/src/components/VisualMatchingPairs/VisualMatchingPairs.js

```diff
@@ -0,0 +1,267 @@
+import React, { useEffect, useRef, useState } from "react";
```

Comment:
This component reimplements a lot of functionality from the `MatchingPairs` component. What functionality is actually different, other than that play cards look different, and that the effect of a click is different? I would be nice to make this DRYer.

Reply:
I tried to use the existing `MatchingPairs` component and extend it but it got quite convoluted because the existing component is built with the `Playback` and audio playback in mind.